features/collection/persistence-rrd/src/main/java/org/opennms/netmgt/collection/persistence/rrd/RrdPersistOperationBuilder.java

```diff
@@ -211,7 +222,9 @@ public void commit() throws PersistException {
             List<RrdDataSource> dataSources = getDataSources();
             if (dataSources != null && dataSources.size() > 0) {
                 createRRD(m_rrdStrategy, ownerName, absolutePath, m_rrdName, getRepository().getStep(), dataSources, getRepository().getRraList(), m_metaData);
-                updateRRD(m_rrdStrategy, ownerName, absolutePath, m_rrdName, m_timeKeeper.getCurrentTime(), getValues());
+                long rrdTime = Boolean.getBoolean("org.opennms.netmgt.collection.persistence.rrd.useCollectionStartTime")
+                    ? m_collectionTime : m_timeKeeper.getCurrentTime();
```

Comment:
If I'm reading the code correctly, it looks like the best way to go about this is to add a setTimeKeeper method to CollectionSet and have all classes that implement CollectionSet set the TimeKeeper on their resources.  We can then create a constant timekeeper in CollectableSerivce with the collection time there and have that be the timekeeper we set on all resources via the previously mentioned method.